### PR TITLE
Add assertExactEquals function to test specification

### PR DESCRIPTION
**Pull request by Dilhasha**
## Purpose
Add assertExactEquals function to test specification. This is applicable only to the "Swan Lake" release.

### CODE REVIEW

File: test/test-framework-spec.md [hunk 1]
```diff
@@ -154,6 +154,18 @@ The following are the list of available assertions available in the test framewo
    <td>Checks if the specified value is not equal to the expected value.
    </td>
   </tr>
+  <tr>
+     <td>@test:assertExactEquals
+     </td>
+     <td>Checks if the specified entity is exactly equal to the expected entity.
```
Comment (azinneera) [inline test/test-framework-spec.md:160]: Should we make this more clear saying both the values refer to the same?

Comment (Dilhasha) [reply test/test-framework-spec.md]: Improved function descriptions with b46bdc4

File: test/test-framework-spec.md [hunk 2]
```diff
@@ -154,6 +154,20 @@ The following are the list of available assertions available in the test framewo
...
   </tr>
+  <tr>
+     <td>@test:assertExactEquals
+     </td>
+     <td>Checks if the specified value is exactly equal to the expected value
+     i.e. both refer to the same entity.
+     </td>
+    </tr>
+    <tr>
+     <td>@test:assertExactNotEquals
```
Comment (hevayo) [inline test/test-framework-spec.md:165]: shouldn't this be assertNotExactEquals ?

Comment (Dilhasha) [reply test/test-framework-spec.md:165]: Yes, you are right.

**Pull request merged**